tfs/src/main/java/hudson/plugins/tfs/model/AbstractHookEvent.java

```diff
@@ -294,15 +313,24 @@ private Boolean branchMatches(final GitCodePushedEventArgs gitCodePushedEventArg
                     for (final RemoteConfig repository : git.getRepositories()) {
                         boolean repositoryMatches = false;
                         for (final URIish remoteURL : repository.getURIs()) {
-                            if (UriHelper.areSameGitRepo(uri, remoteURL)) {
+                            String url = this.envVars.expand(remoteURL.toString());

```

Comment:
final?

Reply:
CheckStyle overlooked this case as well -- perhaps the logic around it make it confused whether this var was modified later or not.